File: agent/gemini.js
import { GoogleGenerativeAI } from '@google/generative-ai';
import logger from '../utils/logger.js';

const apiKey = process.env.GEMINI_API_KEY;
if (!apiKey) throw new Error('GEMINI_API_KEY is not set');

const modelName = process.env.GEMINI_MODEL || 'gemini-1.5-flash'; // Actualizado a versión 2025

const genai = new GoogleGenerativeAI(apiKey);
const model = genai.getGenerativeModel({ model: modelName });

export async function generateContent(contents) {
  try {
    const result = await model.generateContent(contents);
    return result.response.text();
  } catch (error) {
    logger.error(`Gemini API error: ${error.message}`);
    throw error;
  }
}

File: agent/gemini.py
from google import genai  # Asumiendo que es google-generativeai o similar, ajusta si es necesario
import os

# Setup de Gemini
google_api_key = os.getenv('GEMINI_API_KEY')
if not google_api_key:
    raise RuntimeError('GEMINI_API_KEY is not set')
model_name = os.getenv('GEMINI_MODEL', 'gemini-2.5-flash')
genai_client = genai.Client(api_key=google_api_key)
